Mathlib/LinearAlgebra/AffineSpace/Independent.lean

```diff
@@ -776,6 +776,31 @@ theorem AffineIndependent.affineIndependent_update_of_notMem_affineSpan [Decidab
   replace ha : AffineIndependent k fun x : {x | x ≠ i} ↦ f x := h₁ ▸ AffineIndependent.subtype ha _
   exact AffineIndependent.affineIndependent_of_notMem_span ha <| by aesop
 
+
+/-- Swapping the first two points preserves affine independence. -/
+theorem AffineIndependent.comm_left {p₁ p₂ p₃ : P} (h : AffineIndependent k ![p₁, p₂, p₃]) :
+    AffineIndependent k ![p₂, p₁, p₃] := by
+  rw [← affineIndependent_equiv (Equiv.swap 0 1)]
+  convert h using 1
+  ext x
+  fin_cases x <;> rfl
+
+/-- Swapping the last two points preserves affine independence. -/
+theorem AffineIndependent.comm_right {p₁ p₂ p₃ : P} (h : AffineIndependent k ![p₁, p₂, p₃]) :
+    AffineIndependent k ![p₁, p₃, p₂] := by
+  rw [← affineIndependent_equiv (Equiv.swap 1 2)]
+  convert h using 1
+  ext x
+  fin_cases x <;> rfl
+
+/-- Reversing the order of three points preserves affine independence. -/
+theorem AffineIndependent.reverse_of_three (p₁ p₂ p₃ : P) (h : AffineIndependent k ![p₁, p₂, p₃]) :
+    AffineIndependent k ![p₃, p₂, p₁] := by
+  rw [← affineIndependent_equiv (Equiv.swap 0 2)]
+  convert h using 1
+  ext x
+  fin_cases x <;> rfl
```

Comment:
I'm not sure we're on the right path if we need lemmas like this. For one thing, there are 6 permutations of three things, so this isn't even the end of the story. For another, what about when we have four points?

I think what we really want is better automation / tactics, similar to how we use `abel` or `ring` rather than having enormously-many lemmas.

I've started a conversation [on Zulip](https://leanprover.zulipchat.com/#narrow/channel/217875-Is-there-code-for-X.3F/topic/Simproc.20for.20permuting.20vector.20components/with/560645869) in the hopes of getting somewhere. Perhaps you can add some motivation from this work to that thread.